src/persistent-hash-map.mjs

```diff
@@ -0,0 +1,948 @@
+import { isEqual } from "./gleam.mjs";
```

Comment:
What was used as a reference for this module?

Reply:
The assoc/find/without methods are based on
https://github.com/clojure/clojure/blob/master/src/jvm/clojure/lang/PersistentHashMap.java
Ended up having quite a few deviations though (no interleaved key-value array for one, and written in a more functional style).
The hash function is mostly made based on `isEqual` in prelude such that `isEqual(a,b)` implies `hash(a) == hash(b)`. Some inspiration from immutable.js.

Reply:
Wonderful, thank you.